### PR TITLE
Task #1.1 Introducing Middleware and Unit Testing

### PR DESCRIPTION
**Pull request by exemory**

### CODE REVIEW

File: Business.Tests/CommentServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+        _unitOfWorkMock.Setup(u => u.GameRepository.GetByIdAsync(commentCreationDto.GameId))
+            .ReturnsAsync(_fixture.Create<Game>());
+        _unitOfWorkMock.Setup(u => u.CommentRepository.Add(It.IsAny<Comment>()));
+
+        var result = await _sut.CreateAsync(commentCreationDto);
+
+        result.Should().BeEquivalentTo(expected);
+
+        _unitOfWorkMock.Verify(u => u.CommentRepository.Add(It.Is<Comment>(g => expectedToCreate.Equals(g))),
```
Comment (ataromoku) [inline Business.Tests/CommentServiceTests.cs:48]: You can just do `u.CommentRepository.Add(expectedToCreate)`. Delegate for more complicated rules

File: Business.Tests/CommentServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    {
+        var commentCreationDto = _fixture.Create<CommentCreationDto>();
+
+        _unitOfWorkMock.Setup(u => u.GameRepository.GetByIdAsync(commentCreationDto.GameId))
+            .ReturnsAsync((Game?) null);
+
+        Func<Task> result = async () => await _sut.CreateAsync(commentCreationDto);
+
+        await result.Should().ThrowAsync<NotFoundException>()
+            .WithMessage($"Game with id '{commentCreationDto.GameId}' not found.");
```
Comment (ataromoku) [inline Business.Tests/CommentServiceTests.cs:89]: Extract string to local constant

**Review by ataromoku — CHANGES_REQUESTED** [review]

File: Business.Tests/CommentServiceTests.cs (hunk 1 above)
Comment (exemory) [reply Business.Tests/CommentServiceTests.cs:48]: I can't do this because expectedToCreate is of type ExpectedObject

Comment (ataromoku) [reply Business.Tests/CommentServiceTests.cs:48]: You're right

**Review by ataromoku — APPROVED** [review]

**Pull request merged**